File: 28278/28278.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   28278.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2024/02/17 13:16:02 by jihoolee          #+#    #+#             */
/*   Updated: 2024/02/17 16:29:34 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <stack>

int pop_stack(std::stack<int>* s) {
  if (s->empty())
    return -1;
  int result = s->top();

  s->pop();
  return result;
}

void print_top(std::stack<int>* s) {
  if (s->empty())
    std::cout << -1 << '\n';
  else
    std::cout << s->top() << '\n';
}

void do_cmd(std::stack<int>* s, int cmd) {
  std::stack<int>&  r_s = *s;
  int               num;

  switch (cmd) {
    case 1:
      std::cin >> num;
      s->push(num);
      break;
    case 2:
      std::cout << pop_stack(s) << '\n';
      break;
    case 3:
      std::cout << s->size() << '\n';
      break;
    case 4:
      std::cout << int(s->empty()) << '\n';
      break;
    case 5:
      print_top(s);
      break;
  }
}

int main(void) {
  std::ios::sync_with_stdio(false);
  std::cin.tie(NULL);
  std::cout.tie(NULL);

  int             cmd_cnt;
  int             cmd;
  std::stack<int> s;

  std::cin >> cmd_cnt;
  for (int i = 0; i < cmd_cnt; ++i) {
    std::cin >> cmd;
    do_cmd(&s, cmd);
  }
  return 0;
}

File: 28278/28278.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    28278.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2024/02/17 15:20:06 by jihoolee          #+#    #+#              #
#    Updated: 2024/02/17 16:49:09 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

import sys


class Stack:
    def __init__(self):
        self.container: list[int] = []
        self.size: int = 0

    def push(self, num: int) -> None:
        self.container.append(num)
        self.size += 1

    def pop(self) -> int:
        if (self.empty()):
            return -1
        result: int = self.container.pop()
        self.size -= 1
        return result

    def empty(self) -> bool:
        if (self.size == 0):
            return True
        else:
            return False

    def top(self) -> int:
        if (self.empty()):
            return -1
        return self.container[-1]


def main() -> None:
    s: Stack = Stack()

    for _ in range(int(input())):
        cmd: list[int] = list(map(int, sys.stdin.readline().strip().split()))

        if cmd[0] == 1:
            s.push(cmd[1])
        elif cmd[0] == 2:
            print(s.pop())
        elif cmd[0] == 3:
            print(s.size)
        elif cmd[0] == 4:
            print(int(s.empty()))
        elif cmd[0] == 5:
            print(s.top())
        else:
            raise ValueError('Command can must be number between 1-5')


if __name__ == '__main__':
    main()
